File: public/examples/integrations/toolkits/web/crawl_website_example_llm_oai.py
from arcadepy import Arcade

USER_ID = "you@example.com"
TOOL_NAME = "Web.CrawlWebsite"

client = Arcade()  # Automatically finds the `ARCADE_API_KEY` env variable

tool_input = {"url": "https://example.com", "max_depth": 2}

response = client.tools.execute(
    tool_name=TOOL_NAME,
    input=tool_input,
    user_id=USER_ID,
)
print(response)

File: public/examples/integrations/toolkits/web/crawl_website_example_llm_oai.js
import OpenAI from 'openai';

const USER_ID = 'you@example.com';
const PROMPT = 'Crawl the website https://example.com with a maximum depth of 2.';
const TOOL_NAME = 'Web.CrawlWebsite';

const client = new OpenAI({
  baseURL: 'https://api.arcade.dev',
  apiKey: process.env.ARCADE_API_KEY
});

const response = await client.chat.completions.create({
  messages: [
    { role: 'user', content: PROMPT }
  ],
  model: 'gpt-4o-mini',
  user: USER_ID,
  tools: [TOOL_NAME],
  tool_choice: 'generate'
});

console.log(response.choices[0].message.content); 